pkg/crc/config/config.go

```diff
@@ -95,6 +96,36 @@ func (c *Config) Set(key string, value interface{}) (string, error) {
 		return "", fmt.Errorf(invalidType, value, key)
 	}
 
+	// Preset is mapped with `memory`, `cpus` and `bundle` and
+	// we want to make sure if cpu or memory is less for a preset
+	// then default is set automatic.
+	if setting.Name == Preset {
+		preset := preset.ParsePreset(value.(string))
+		mem := c.Get(Memory)
+		if err := validation.ValidateMemory(mem.AsInt(), preset); err != nil {
+			if _, err := c.Unset(Memory); err != nil {
+				return "", err
+			}
+		}
+		cpu := c.Get(CPUs)
+		if err := validation.ValidateCPUs(cpu.AsInt(), preset); err != nil {
```

Comment:
shouldn't we also use disk-size? This is now identical, but would suggest to create an issue to follow-up.